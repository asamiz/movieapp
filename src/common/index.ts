export * from './colors';
export * from './fonts';
export * from './styles';
export * from './interfaces';
export * from './apiConfig';
export * from './static';
