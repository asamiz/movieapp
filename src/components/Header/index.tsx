import React from 'react';
import { Text, View } from 'react-native';
import styles from './styles';

interface Props {
	text: string;
	containerStyle?: object;
	textStyle?: object;
}

const Header = ({ text, containerStyle, textStyle }: Props) => (
	<View style={containerStyle} testID={'app-header-container'}>
		<Text style={[styles.text, textStyle]} testID={'app-header-text'}>
			{text}
		</Text>
	</View>
);

export { Header };
