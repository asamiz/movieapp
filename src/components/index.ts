export * from './Header';
export * from './MovieCard';
export * from './Rating';
export * from './SearchBar';
