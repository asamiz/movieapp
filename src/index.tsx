import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const App = () => (
  <View style={styles.container}>
    <Text>{'Movie App '}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default App;
