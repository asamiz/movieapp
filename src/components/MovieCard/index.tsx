import React from 'react';
import { Text, View, Pressable } from 'react-native';
import FastImage from 'react-native-fast-image';
import { DUMMY_IMAGE, IShortMovie } from '../../common';
import styles from './styles';

const MovieCard = ({
 Title, Poster, Year, onPress,
}: IShortMovie) => (
	<Pressable style={styles.container} onPress={onPress}>
		<View style={styles.contentContainer}>
			<Text style={styles.title}>{Title}</Text>
			<Text style={styles.bodyText}>{`Released: ${Year}`}</Text>
		</View>
		<FastImage
			source={{ uri: Poster === 'N/A' ? DUMMY_IMAGE : Poster }}
			resizeMode={'stretch'}
			style={styles.image}
		/>
	</Pressable>
);

export { MovieCard };
