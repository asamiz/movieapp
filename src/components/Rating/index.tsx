import React from 'react';
import { Text, View } from 'react-native';
import StarRating from 'react-native-star-rating';
import { COLORS } from '../../common';
import styles from './styles';

interface Props {
	rating: number;
}

const Rating = ({ rating }: Props) => (
	<View style={styles.ratingContainer}>
		<StarRating
			disabled={true}
			maxStars={5}
			rating={rating - 5}
			starSize={12}
			emptyStar={'star'}
			fullStar={'star'}
			halfStar={'star-half'}
			emptyStarColor={COLORS.independence}
			fullStarColor={COLORS.sun}
			starStyle={styles.star}
		/>
		<Text style={styles.rating}>{`${rating} / 10`}</Text>
	</View>
);

export { Rating };
