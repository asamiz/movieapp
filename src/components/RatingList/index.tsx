import React from 'react';
import { View, Text } from 'react-native';
import { IRates } from '../../common';
import styles from './styles';

interface Props {
	rates?: any;
}

const RatingList = ({ rates }: Props) => (
	<View style={styles.container} testID={'app-rating-list'}>
		{rates?.map((item: IRates, index: number) => (
			<View key={item.Source + 1} style={styles.col}>
				<Text style={styles.source}>{index === 0 ? 'IMDB' : item.Source}</Text>
				<Text style={styles.value}>{item.Value}</Text>
			</View>
		))}
	</View>
);

export { RatingList };
