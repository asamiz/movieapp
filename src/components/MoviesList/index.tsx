import { useNavigation } from '@react-navigation/native';
import React, { ElementType } from 'react';
import { FlatList } from 'react-native';
import { useDispatch } from 'react-redux';
import { IShortMovie } from '../../common';
import { storeMovie } from '../../store/actions';
import { MovieCard } from '../MovieCard';
import styles from './styles';

interface Props {
	data: IShortMovie[];
	ListEmptyComponent: ElementType;
	ListHeaderComponent: ElementType;
}

const MoviesList = ({
	data,
	ListEmptyComponent,
	ListHeaderComponent,
}: Props) => {
	const navigation = useNavigation();
	const dispatch = useDispatch();

	const onPressCard = (movie: IShortMovie) => {
		dispatch(storeMovie(movie));
		navigation.navigate('MovieScreen', { id: movie.imdbID });
	};

	return (
		<FlatList
			testID={'app-movie-list'}
			data={data}
			keyboardShouldPersistTaps={'always'}
			renderItem={({ item }) => (
				<MovieCard
					Title={item.Title}
					imdbID={item.imdbID}
					Poster={item.Poster}
					Year={item.Year}
					onPress={() => onPressCard(item)}
				/>
			)}
			ListEmptyComponent={<ListEmptyComponent />}
			ListHeaderComponent={<ListHeaderComponent />}
			contentContainerStyle={styles.contentContainer}
			keyExtractor={(item) => item.imdbID}
			showsVerticalScrollIndicator={false}
		/>
	);
};

export { MoviesList };
