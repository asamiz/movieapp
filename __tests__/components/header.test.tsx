import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Header } from '../../src/components';
import { render } from 'react-native-testing-library';

describe('App Header Unit Testing', () => {
	/** Test component rendering */
	it('should render correctly', () => {
		const tree = TestRenderer.create(<Header text={'CARA CARE'} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	/** Test required passed values */
	it('should render text prop properly', () => {
		const { getByTestId } = render(<Header text={'Test'} />);
		const element = getByTestId('app-header-text');
		expect(element.children[0]).toBe('Test');
	});
});
