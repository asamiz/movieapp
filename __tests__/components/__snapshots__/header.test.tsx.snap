// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Header Unit Testing should render correctly 1`] = `
<View
  testID="app-header-container"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 44.5,
        },
        undefined,
      ]
    }
    testID="app-header-text"
  >
    CARA CARE
  </Text>
</View>
`;
